plugin/src/ast_helpers/number.rs

```diff
@@ -92,6 +94,16 @@ impl PartialOrd for Number {
     }
 }
 
+impl hash::Hash for Number {
+    fn hash<H: hash::Hasher>(&self, state: &mut H) {
+        match self {
+            Number::NaN => "NaN".hash(state),
+            Number::Integer(value) => (*value as f64).to_bits().hash(state),
```

Comment:
I really don't know what I'm doing here as far as a hash implementation but what I ran into (which sounded vaguely familiar) was that before I had `value.hash(state)` here but then was seeing some test cases fail I think because basically two `Number`'s that would compare equal weren't hashing to the same value